### ISSUE TITLE
Validation crashes when continuing training on a previous checkpoint

### ISSUE THREAD

**Issue by evellasques** [aws-neuron/neuronx-nemo-megatron#16]
Background:

When `val_check_interval` is set to `1`, validation loss works as expected. However, when one loads a previous checkpoint (by setting `+model.load_xser=True` and +model.resume_from_checkpoint=`/shared/model/mp_rank_00/megatron_bigcode--step\=154-consumed_samples\=156672.0.ckpt`), `MegatronGPTModel.validation_epoch_end()` is crashing on line 805 https://github.com/aws-neuron/neuronx-nemo-megatron/blob/6038bb80b5ae59d3f4a99ed38a4f68fa0e22665a/nemo/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py#L805
this is the error message:
```
averaged_loss = torch.stack(outputs).mean()
  File "/shared/home/env/aws_neuron_venv_pytorch_py39v2/lib/python3.9/site-packages/pytorch_lightning/loops/dataloader/evaluation_loop.py", li
ne 180, in on_run_end
  File "/shared/home/env/aws_neuron_venv_pytorch_py39v2/lib/python3.9/site-packages/hydra/_internal/utils.py", line 213, in run_and_report
  File "/shared/home/env/aws_neuron_venv_pytorch_py39v2/lib/python3.9/site-packages/nemo/collections/nlp/models/language_modeling/megatron_gpt
_model.py", line 805, in validation_epoch_end
RuntimeError: stack expects a non-empty TensorList
RuntimeError: stack expects a non-empty TensorList
    self._evaluation_epoch_end(self._outputs)
  File "/shared/home/env/aws_neuron_venv_pytorch_py39v2/lib/python3.9/site-packages/pytorch_lightning/loops/dataloader/evaluation_loop.py", li
ne 288, in _evaluation_epoch_end

  File "/shared/home/env/aws_neuron_venv_pytorch_py39v2/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1200, in _run_
train
        trainer.fit(model)    
self._outputs = self.epoch_loop.run(self._data_fetcher)self._run(model, ckpt_path=self.ckpt_path)
```

Basically, `outputs` is an empty list. But for what I understood, in that particular case line 802 `parallel_state.is_pipeline_last_stage()` should return `False`.

**Comment by jyang-aws** [comment]
@evellasques thanks for filing the issue. we're looking into it.

**Comment by jyang-aws** [comment]
could you give a try by changing 
```
+model.resume_from_checkpoint=/shared/model/mp_rank_00/megatron_bigcode--step\=154-consumed_samples\=156672.0.ckpt
```
to this:
```
+model.resume_from_checkpoint=/shared/model/megatron_bigcode--step\=154-consumed_samples\=156672.0.ckpt
```
basically, making sure the checkpoint path is valid.
